src/acquisition/flusurv/flusurv_update.py

```diff
@@ -175,18 +321,28 @@ def main():
     # fmt: on
     args = parser.parse_args()
 
+    metadata = flusurv.fetch_flusurv_metadata()
+
     # scrape current issue from the main page
-    issue = flusurv.get_current_issue()
+    issue = flusurv.get_current_issue(metadata)
     print(f"current issue: {int(issue)}")
 
+    # Ignore seasons with all dates older than one year
+    seasonids = {
+        season_blob["seasonid"] for season_blob in metadata["seasons"]
+        if delta_epiweeks(flusurv.mmwrid_to_epiweek(season_blob["endweek"]), issue) < max_age_to_consider_weeks
+    }
```

Comment:
this is kind of a tricky, and maybe not worth the effort, but.......

does it make sense to push some of this functionality into `update()` or (more preferably) even further into `flusurv.get_data()`?

when this module has lots of intricate calls of specific stuff from `flusurv.py`, it loses its [cohesion](https://en.wikipedia.org/wiki/Cohesion_(computer_science)#/media/File:CouplingVsCohesion.svg)  ...and then we might just as well combine the modules `flusurv.py` and `flusurv_update.py`.

i understand trying to make only one network call to retrieve `metadata` -- if youre worried about reprocessing stuff found in `metadata`, you could do something like making a `FlusurvMetadata` class that precomputes `issue` and `seasonids` and (from `flusurv.py`) `id_label_map` and `id_season_map` once each, and then pass an instance of that around.

Reply:
Thanks for mentioning cohesion, I hadn't heard of that concept before. I created a metadata class like you suggested and also a data-fetcher class that uses the metadata one. Only the fetcher class is called from the wrapper/main file.